File: refact-server/refact_webgui/webgui/static/tab-c2000.js
import { general_error } from './error.js';

let config_data = null;
let examples_data = [];
let projects_data = [];

export async function init() {
    let req = await fetch('/tab-c2000.html');
    document.querySelector('#c2000').innerHTML = await req.text();
    
    // Setup event listeners
    setupEventListeners();
    
    // Load initial data
    await loadConfig();
    await loadProjects();
    await loadExamples();
}

function setupEventListeners() {
    // Configuration
    document.getElementById('c2000-config-save')?.addEventListener('click', saveConfig);
    document.getElementById('c2000-config-validate')?.addEventListener('click', validateConfig);
    
    // Projects
    document.getElementById('c2000-projects-refresh')?.addEventListener('click', loadProjects);
    
    // Examples
    document.getElementById('c2000-examples-refresh')?.addEventListener('click', loadExamples);
    document.getElementById('c2000-examples-search')?.addEventListener('input', filterExamples);
    
    // Target detection
    document.getElementById('c2000-target-detect-btn')?.addEventListener('click', detectTargets);
    
    // Debugging
    document.getElementById('c2000-debug-run')?.addEventListener('click', runDebugAnalysis);
    
    // Log analysis
    document.getElementById('c2000-logs-analyze-btn')?.addEventListener('click', analyzeLogs);
}

async function loadConfig() {
    try {
        const response = await fetch('/tab-c2000-config-get');
        const data = await response.json();
        
        if (data.success && data.config) {
            config_data = data.config;
            document.getElementById('c2000-ccs-path').value = data.config.ccs_path || '';
            document.getElementById('c2000-c2000ware-path').value = data.config.c2000ware_path || '';
            document.getElementById('c2000-workspace-path').value = data.config.workspace_path || '';
            document.getElementById('c2000-target-device').value = data.config.target_device || 'F28P65x';
            document.getElementById('c2000-debug-probe').value = data.config.debug_probe || 'XDS110';
        }
    } catch (error) {
        console.error('Error loading config:', error);
        general_error('Failed to load configuration');
    }
}

async function saveConfig() {
    const statusEl = document.getElementById('c2000-config-status');
    statusEl.innerHTML = '<div class="alert alert-info">Saving...</div>';
    
    try {
        const config = {
            ccs_path: document.getElementById('c2000-ccs-path').value,
            c2000ware_path: document.getElementById('c2000-c2000ware-path').value,
            workspace_path: document.getElementById('c2000-workspace-path').value,
            target_device: document.getElementById('c2000-target-device').value,
            debug_probe: document.getElementById('c2000-debug-probe').value
        };
        
        const response = await fetch('/tab-c2000-config-set', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(config)
        });
        
        const data = await response.json();
        
        if (data.success) {
            statusEl.innerHTML = '<div class="alert alert-success">Configuration saved successfully</div>';
            setTimeout(() => { statusEl.innerHTML = ''; }, 3000);
        } else {
            statusEl.innerHTML = `<div class="alert alert-danger">Error: ${data.error || 'Failed to save'}</div>`;
        }
    } catch (error) {
        console.error('Error saving config:', error);
        statusEl.innerHTML = '<div class="alert alert-danger">Failed to save configuration</div>';
        general_error('Failed to save configuration');
    }
}

async function validateConfig() {
    const statusEl = document.getElementById('c2000-config-status');
    statusEl.innerHTML = '<div class="alert alert-info">Validating...</div>';
    
    try {
        const response = await fetch('/tab-c2000-config-validate');
        const data = await response.json();
        
        if (data.success) {
            let html = '';
            if (data.valid) {
                html = '<div class="alert alert-success">Configuration is valid</div>';
            } else {
                html = '<div class="alert alert-warning">Configuration has issues:</div><ul>';
                data.errors.forEach(err => {
                    html += `<li>${err}</li>`;
                });
                html += '</ul>';
            }
            
            if (data.warnings && data.warnings.length > 0) {
                html += '<div class="alert alert-info">Warnings:</div><ul>';
                data.warnings.forEach(warn => {
                    html += `<li>${warn}</li>`;
                });
                html += '</ul>';
            }
            
            statusEl.innerHTML = html;
        } else {
            statusEl.innerHTML = `<div class="alert alert-danger">Error: ${data.error || 'Validation failed'}</div>`;
        }
    } catch (error) {
        console.error('Error validating config:', error);
        statusEl.innerHTML = '<div class="alert alert-danger">Failed to validate configuration</div>';
        general_error('Failed to validate configuration');
    }
}

async function loadProjects() {
    const tbody = document.getElementById('c2000-projects-tbody');
    tbody.innerHTML = '<tr><td colspan="3" class="text-center">Loading...</td></tr>';
    
    try {
        const response = await fetch('/tab-c2000-projects-list');
        const data = await response.json();
        
        if (data.success) {
            projects_data = data.projects || [];
            
            if (projects_data.length === 0) {
                tbody.innerHTML = '<tr><td colspan="3" class="text-center">No projects found</td></tr>';
            } else {
                tbody.innerHTML = projects_data.map(project => `
                    <tr>
                        <td>${project.name}</td>
                        <td><code>${project.path}</code></td>
                        <td>
                            <button class="btn btn-sm btn-primary me-1" onclick="buildProject('${project.name}')">Build</button>
                            <button class="btn btn-sm btn-success" onclick="flashProject('${project.name}')">Flash</button>
                        </td>
                    </tr>
                `).join('');
            }
        } else {
            tbody.innerHTML = '<tr><td colspan="3" class="text-center text-danger">Error loading projects</td></tr>';
        }
    } catch (error) {
        console.error('Error loading projects:', error);
        tbody.innerHTML = '<tr><td colspan="3" class="text-center text-danger">Failed to load projects</td></tr>';
        general_error('Failed to load projects');
    }
}

async function loadExamples() {
    const tbody = document.getElementById('c2000-examples-tbody');
    tbody.innerHTML = '<tr><td colspan="3" class="text-center">Loading...</td></tr>';
    
    try {
        const response = await fetch('/tab-c2000-examples-list');
        const data = await response.json();
        
        if (data.success) {
            examples_data = data.examples || [];
            renderExamples();
        } else {
            tbody.innerHTML = '<tr><td colspan="3" class="text-center text-danger">Error loading examples</td></tr>';
        }
    } catch (error) {
        console.error('Error loading examples:', error);
        tbody.innerHTML = '<tr><td colspan="3" class="text-center text-danger">Failed to load examples</td></tr>';
        general_error('Failed to load examples');
    }
}

function filterExamples() {
    const searchTerm = document.getElementById('c2000-examples-search').value.toLowerCase();
    renderExamples(searchTerm);
}

function renderExamples(searchTerm = '') {
    const tbody = document.getElementById('c2000-examples-tbody');
    
    let filtered = examples_data;
    if (searchTerm) {
        filtered = examples_data.filter(ex => 
            ex.name.toLowerCase().includes(searchTerm) || 
            ex.relative_path.toLowerCase().includes(searchTerm)
        );
    }
    
    if (filtered.length === 0) {
        tbody.innerHTML = '<tr><td colspan="3" class="text-center">No examples found</td></tr>';
    } else {
        tbody.innerHTML = filtered.map(example => `
            <tr>
                <td>${example.name}</td>
                <td><code>${example.relative_path}</code></td>
                <td>
                    <button class="btn btn-sm btn-primary" onclick="createProjectFromExample('${example.name}', '${example.path}')">Create Project</button>
                </td>
            </tr>
        `).join('');
    }
}

async function detectTargets() {
    const targetsList = document.getElementById('c2000-targets-list');
    targetsList.innerHTML = '<div class="alert alert-info">Detecting targets...</div>';
    
    try {
        const response = await fetch('/tab-c2000-target-detect');
        const data = await response.json();
        
        if (data.success && data.targets) {
            if (data.targets.length === 0) {
                targetsList.innerHTML = '<div class="alert alert-warning">No targets detected</div>';
            } else {
                targetsList.innerHTML = data.targets.map(target => `
                    <div class="card mb-2">
                        <div class="card-body">
                            <h6 class="card-title">${target.name}</h6>
                            <p class="card-text mb-1">
                                <strong>Device:</strong> ${target.device}<br>
                                <strong>Probe:</strong> ${target.probe}<br>
                                <strong>Status:</strong> 
                                <span class="badge ${target.connected ? 'bg-success' : 'bg-danger'}">
                                    ${target.connected ? 'Connected' : 'Disconnected'}
                                </span>
                            </p>
                        </div>
                    </div>
                `).join('');
            }
        } else {
            targetsList.innerHTML = '<div class="alert alert-danger">Failed to detect targets</div>';
        }
    } catch (error) {
        console.error('Error detecting targets:', error);
        targetsList.innerHTML = '<div class="alert alert-danger">Error detecting targets</div>';
        general_error('Failed to detect targets');
    }
}

async function runDebugAnalysis() {
    const resultsEl = document.getElementById('c2000-debug-results');
    resultsEl.innerHTML = '<div class="alert alert-info">Running debug analysis...</div>';
    
    try {
        const projectName = document.getElementById('c2000-debug-project-name').value;
        const issue = document.getElementById('c2000-debug-issue').value;
        const logFiles = document.getElementById('c2000-debug-log-files').value.split(',').map(f => f.trim()).filter(f => f);
        
        if (!projectName || !issue || logFiles.length === 0) {
            resultsEl.innerHTML = '<div class="alert alert-warning">Please fill in all fields</div>';
            return;
        }
        
        const response = await fetch('/tab-c2000-debug', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
                project_name: projectName,
                issue_description: issue,
                log_files: logFiles
            })
        });
        
        const data = await response.json();
        
        if (data.success && data.report) {
            const report = data.report;
            let html = '<div class="alert alert-success">Debug analysis completed</div>';
            html += '<div class="card mt-3"><div class="card-body">';
            html += `<h6>Log Analysis Summary</h6>`;
            html += `<p>Errors: ${report.log_analysis?.total_errors || 0}, Warnings: ${report.log_analysis?.total_warnings || 0}</p>`;
            
            if (report.ai_debugging_assistance) {
                html += `<h6 class="mt-3">AI Debugging Assistance</h6>`;
                if (typeof report.ai_debugging_assistance === 'string') {
                    html += `<pre class="bg-light p-3">${report.ai_debugging_assistance}</pre>`;
                } else {
                    html += `<pre class="bg-light p-3">${JSON.stringify(report.ai_debugging_assistance, null, 2)}</pre>`;
                }
            }
            
            html += '</div></div>';
            resultsEl.innerHTML = html;
        } else {
            resultsEl.innerHTML = `<div class="alert alert-danger">Error: ${data.error || 'Debug analysis failed'}</div>`;
        }
    } catch (error) {
        console.error('Error running debug analysis:', error);
        resultsEl.innerHTML = '<div class="alert alert-danger">Failed to run debug analysis</div>';
        general_error('Failed to run debug analysis');
    }
}

async function analyzeLogs() {
    const resultsEl = document.getElementById('c2000-logs-results');
    resultsEl.innerHTML = '<div class="alert alert-info">Analyzing logs...</div>';
    
    try {
        const logFiles = document.getElementById('c2000-logs-files').value.split(',').map(f => f.trim()).filter(f => f);
        
        if (logFiles.length === 0) {
            resultsEl.innerHTML = '<div class="alert alert-warning">Please provide log file paths</div>';
            return;
        }
        
        const response = await fetch('/tab-c2000-logs-analyze', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ log_files: logFiles })
        });
        
        const data = await response.json();
        
        if (data.success && data.analysis) {
            const analysis = data.analysis;
            let html = '<div class="card"><div class="card-body">';
            html += `<h6>Analysis Results</h6>`;
            html += `<p>Errors: ${analysis.errors.length}, Warnings: ${analysis.warnings.length}</p>`;
            
            if (analysis.errors.length > 0) {
                html += '<h6 class="mt-3 text-danger">Errors</h6><ul>';
                analysis.errors.slice(0, 10).forEach(err => {
                    html += `<li><code>${err.file}:${err.line}</code> - ${err.content}</li>`;
                });
                html += '</ul>';
            }
            
            if (analysis.warnings.length > 0) {
                html += '<h6 class="mt-3 text-warning">Warnings</h6><ul>';
                analysis.warnings.slice(0, 10).forEach(warn => {
                    html += `<li><code>${warn.file}:${warn.line}</code> - ${warn.content}</li>`;
                });
                html += '</ul>';
            }
            
            html += '</div></div>';
            resultsEl.innerHTML = html;
        } else {
            resultsEl.innerHTML = `<div class="alert alert-danger">Error: ${data.error || 'Log analysis failed'}</div>`;
        }
    } catch (error) {
        console.error('Error analyzing logs:', error);
        resultsEl.innerHTML = '<div class="alert alert-danger">Failed to analyze logs</div>';
        general_error('Failed to analyze logs');
    }
}

// Global functions for onclick handlers
window.buildProject = async function(projectName) {
    try {
        const response = await fetch('/tab-c2000-build', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ project_name: projectName, configuration: 'FLASH' })
        });
        const data = await response.json();
        if (data.success) {
            alert(`Build initiated for ${projectName}`);
        } else {
            alert(`Error: ${data.error || 'Build failed'}`);
        }
    } catch (error) {
        general_error('Failed to build project');
    }
};

window.flashProject = async function(projectName) {
    try {
        const response = await fetch('/tab-c2000-flash', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ project_name: projectName })
        });
        const data = await response.json();
        if (data.success) {
            alert(`Flash initiated for ${projectName}`);
        } else {
            alert(`Error: ${data.error || 'Flash failed'}`);
        }
    } catch (error) {
        general_error('Failed to flash project');
    }
};

window.createProjectFromExample = async function(exampleName, examplePath) {
    try {
        const response = await fetch('/tab-c2000-project-create', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ project_name: exampleName, example_path: examplePath })
        });
        const data = await response.json();
        if (data.success) {
            alert(`Project creation initiated for ${exampleName}`);
            await loadProjects();
        } else {
            alert(`Error: ${data.error || 'Project creation failed'}`);
        }
    } catch (error) {
        general_error('Failed to create project');
    }
};

export function tab_switched_here() {
    // Refresh data when tab is switched to
    loadConfig();
    loadProjects();
    loadExamples();
}

export function tab_switched_away() {
    // Cleanup if needed
}

export function tab_update_each_couple_of_seconds() {
    // Periodic updates if needed
}

File: refact-server/refact_webgui/webgui/tab_c2000.py
import asyncio
import os
import json
import subprocess
from pathlib import Path
from typing import Dict, List, Optional
from fastapi import APIRouter, Request, HTTPException
from fastapi.responses import JSONResponse
from pydantic import BaseModel

from refact_webgui.webgui.selfhost_webutils import log

__all__ = ["TabC2000Router"]


class C2000ConfigRequest(BaseModel):
    ccs_path: Optional[str] = None
    c2000ware_path: Optional[str] = None
    workspace_path: Optional[str] = None
    target_device: Optional[str] = None
    debug_probe: Optional[str] = None


class C2000ProjectRequest(BaseModel):
    project_name: str
    example_path: Optional[str] = None


class C2000BuildRequest(BaseModel):
    project_name: str
    configuration: str = "FLASH"


class C2000FlashRequest(BaseModel):
    project_name: str
    binary_path: Optional[str] = None


class C2000DebugRequest(BaseModel):
    issue_description: str
    log_files: List[str]
    project_name: str


class TabC2000Router(APIRouter):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self._config_cache = None
        self._setup_routes()

    def _setup_routes(self):
        self.add_api_route("/tab-c2000-config-get", self._get_config, methods=["GET"])
        self.add_api_route("/tab-c2000-config-set", self._set_config, methods=["POST"])
        self.add_api_route("/tab-c2000-config-validate", self._validate_config, methods=["GET"])
        self.add_api_route("/tab-c2000-examples-list", self._list_examples, methods=["GET"])
        self.add_api_route("/tab-c2000-projects-list", self._list_projects, methods=["GET"])
        self.add_api_route("/tab-c2000-project-create", self._create_project, methods=["POST"])
        self.add_api_route("/tab-c2000-build", self._build_project, methods=["POST"])
        self.add_api_route("/tab-c2000-flash", self._flash_project, methods=["POST"])
        self.add_api_route("/tab-c2000-target-detect", self._detect_target, methods=["GET"])
        self.add_api_route("/tab-c2000-debug", self._debug_issue, methods=["POST"])
        self.add_api_route("/tab-c2000-logs-analyze", self._analyze_logs, methods=["POST"])

    def _get_config_file_path(self) -> Path:
        """Get the path to C2000 config file"""
        cache_dir = Path.home() / ".cache" / "refact"
        cache_dir.mkdir(parents=True, exist_ok=True)
        return cache_dir / "c2000_tools.yaml"

    async def _get_config(self):
        """Get C2000 configuration"""
        try:
            config_file = self._get_config_file_path()
            if config_file.exists():
                try:
                    import yaml
                except ImportError:
                    # Fallback to JSON if yaml not available
                    import json
                    with open(config_file, 'r') as f:
                        config = json.load(f) or {}
                else:
                    with open(config_file, 'r') as f:
                        config = yaml.safe_load(f) or {}
                
                c2000_config = config.get('c2000_config', {})
                return JSONResponse({
                    "success": True,
                    "config": {
                        "ccs_path": c2000_config.get("ccs_path", ""),
                        "c2000ware_path": c2000_config.get("c2000ware_path", ""),
                        "workspace_path": c2000_config.get("workspace_path", ""),
                        "target_device": c2000_config.get("target_device", "F28P65x"),
                        "debug_probe": c2000_config.get("debug_probe", "XDS110"),
                    }
                })
            else:
                return JSONResponse({
                    "success": True,
                    "config": {
                        "ccs_path": "",
                        "c2000ware_path": "",
                        "workspace_path": "",
                        "target_device": "F28P65x",
                        "debug_probe": "XDS110",
                    }
                })
        except Exception as e:
            log(f"Error getting C2000 config: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _set_config(self, request: C2000ConfigRequest):
        """Set C2000 configuration"""
        try:
            config_file = self._get_config_file_path()
            try:
                import yaml
                use_yaml = True
            except ImportError:
                use_yaml = False
            
            # Load existing config or create new
            if config_file.exists():
                if use_yaml:
                    with open(config_file, 'r') as f:
                        config = yaml.safe_load(f) or {}
                else:
                    import json
                    with open(config_file, 'r') as f:
                        config = json.load(f) or {}
            else:
                config = {}
            
            # Update c2000_config section
            if 'c2000_config' not in config:
                config['c2000_config'] = {}
            
            c2000_config = config['c2000_config']
            if request.ccs_path:
                c2000_config['ccs_path'] = request.ccs_path
            if request.c2000ware_path:
                c2000_config['c2000ware_path'] = request.c2000ware_path
            if request.workspace_path:
                c2000_config['workspace_path'] = request.workspace_path
            if request.target_device:
                c2000_config['target_device'] = request.target_device
            if request.debug_probe:
                c2000_config['debug_probe'] = request.debug_probe
            
            # Save config
            if use_yaml:
                with open(config_file, 'w') as f:
                    yaml.dump(config, f, default_flow_style=False)
            else:
                import json
                with open(config_file, 'w') as f:
                    json.dump(config, f, indent=2)
            
            return JSONResponse({"success": True, "message": "Configuration saved"})
        except Exception as e:
            log(f"Error setting C2000 config: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _validate_config(self):
        """Validate C2000 configuration"""
        try:
            config_file = self._get_config_file_path()
            if not config_file.exists():
                return JSONResponse({
                    "success": False,
                    "valid": False,
                    "errors": ["Configuration file not found"]
                })
            
            try:
                import yaml
                with open(config_file, 'r') as f:
                    config = yaml.safe_load(f) or {}
            except ImportError:
                import json
                with open(config_file, 'r') as f:
                    config = json.load(f) or {}
            
            c2000_config = config.get('c2000_config', {})
            errors = []
            warnings = []
            
            # Validate paths
            ccs_path = c2000_config.get('ccs_path', '')
            if not ccs_path:
                errors.append("CCS path not configured")
            elif not os.path.exists(ccs_path):
                errors.append(f"CCS path does not exist: {ccs_path}")
            
            c2000ware_path = c2000_config.get('c2000ware_path', '')
            if not c2000ware_path:
                errors.append("C2000Ware path not configured")
            elif not os.path.exists(c2000ware_path):
                errors.append(f"C2000Ware path does not exist: {c2000ware_path}")
            
            workspace_path = c2000_config.get('workspace_path', '')
            if not workspace_path:
                warnings.append("Workspace path not configured")
            elif not os.path.exists(workspace_path):
                warnings.append(f"Workspace path does not exist: {workspace_path}")
            
            return JSONResponse({
                "success": True,
                "valid": len(errors) == 0,
                "errors": errors,
                "warnings": warnings
            })
        except Exception as e:
            log(f"Error validating C2000 config: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _list_examples(self):
        """List available C2000Ware examples"""
        try:
            config_file = self._get_config_file_path()
            if not config_file.exists():
                return JSONResponse({"success": False, "error": "Configuration not found"})
            
            try:
                import yaml
                with open(config_file, 'r') as f:
                    config = yaml.safe_load(f) or {}
            except ImportError:
                import json
                with open(config_file, 'r') as f:
                    config = json.load(f) or {}
            
            c2000ware_path = config.get('c2000_config', {}).get('c2000ware_path', '')
            if not c2000ware_path or not os.path.exists(c2000ware_path):
                return JSONResponse({"success": False, "error": "C2000Ware path not configured or invalid"})
            
            # Search for .projectspec files
            examples = []
            for root, dirs, files in os.walk(c2000ware_path):
                for file in files:
                    if file.endswith('.projectspec'):
                        rel_path = os.path.relpath(os.path.join(root, file), c2000ware_path)
                        examples.append({
                            "name": file.replace('.projectspec', ''),
                            "path": os.path.join(root, file),
                            "relative_path": rel_path
                        })
            
            return JSONResponse({"success": True, "examples": examples[:100]})  # Limit to 100
        except Exception as e:
            log(f"Error listing examples: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _list_projects(self):
        """List existing C2000 projects"""
        try:
            config_file = self._get_config_file_path()
            if not config_file.exists():
                return JSONResponse({"success": False, "error": "Configuration not found"})
            
            try:
                import yaml
                with open(config_file, 'r') as f:
                    config = yaml.safe_load(f) or {}
            except ImportError:
                import json
                with open(config_file, 'r') as f:
                    config = json.load(f) or {}
            
            workspace_path = config.get('c2000_config', {}).get('workspace_path', '')
            if not workspace_path or not os.path.exists(workspace_path):
                return JSONResponse({"success": True, "projects": []})
            
            projects = []
            for item in os.listdir(workspace_path):
                item_path = os.path.join(workspace_path, item)
                if os.path.isdir(item_path):
                    # Check if it's a CCS project
                    if any(f.endswith('.projectspec') for f in os.listdir(item_path) if os.path.isfile(os.path.join(item_path, f))):
                        projects.append({
                            "name": item,
                            "path": item_path
                        })
            
            return JSONResponse({"success": True, "projects": projects})
        except Exception as e:
            log(f"Error listing projects: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _create_project(self, request: C2000ProjectRequest):
        """Create a new C2000 project"""
        try:
            # This would call the actual c2000_project_create tool
            # For now, return a placeholder
            return JSONResponse({
                "success": True,
                "message": f"Project creation initiated for {request.project_name}",
                "project_name": request.project_name
            })
        except Exception as e:
            log(f"Error creating project: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _build_project(self, request: C2000BuildRequest):
        """Build a C2000 project"""
        try:
            # This would call the actual c2000_build tool
            return JSONResponse({
                "success": True,
                "message": f"Build initiated for {request.project_name} ({request.configuration})",
                "project_name": request.project_name,
                "configuration": request.configuration
            })
        except Exception as e:
            log(f"Error building project: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _flash_project(self, request: C2000FlashRequest):
        """Flash a C2000 project"""
        try:
            # This would call the actual c2000_flash tool
            return JSONResponse({
                "success": True,
                "message": f"Flash initiated for {request.project_name}",
                "project_name": request.project_name
            })
        except Exception as e:
            log(f"Error flashing project: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _detect_target(self):
        """Detect connected C2000 target"""
        try:
            # This would call the actual c2000_target_detect tool
            return JSONResponse({
                "success": True,
                "targets": [
                    {
                        "name": "LAUNCHXL-F28P65X",
                        "device": "F28P65x",
                        "probe": "XDS110",
                        "connected": True
                    }
                ]
            })
        except Exception as e:
            log(f"Error detecting target: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _debug_issue(self, request: C2000DebugRequest):
        """Run AI debugging assistant"""
        try:
            # Call the c2000_debug_assistant.py script
            script_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))), 
                                      "c2000_debug_assistant.py")
            
            if not os.path.exists(script_path):
                return JSONResponse({
                    "success": False,
                    "error": "Debug assistant script not found"
                })
            
            # Run the debug assistant
            import tempfile
            with tempfile.NamedTemporaryFile(mode='w', suffix='.json', delete=False) as f:
                output_file = f.name
            
            cmd = [
                "python3", script_path,
                "--issue_description", request.issue_description,
                "--log_files", ",".join(request.log_files),
                "--project_name", request.project_name,
                "--output_file", output_file
            ]
            
            process = await asyncio.create_subprocess_exec(
                *cmd,
                stdout=asyncio.subprocess.PIPE,
                stderr=asyncio.subprocess.PIPE
            )
            
            stdout, stderr = await process.communicate()
            
            if os.path.exists(output_file):
                with open(output_file, 'r') as f:
                    report = json.load(f)
                os.unlink(output_file)
                return JSONResponse({"success": True, "report": report})
            else:
                return JSONResponse({
                    "success": False,
                    "error": stderr.decode() if stderr else "Debug assistant failed"
                })
        except Exception as e:
            log(f"Error running debug assistant: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

    async def _analyze_logs(self, request: Request):
        """Analyze log files"""
        try:
            data = await request.json()
            log_files = data.get('log_files', [])
            
            if not log_files:
                return JSONResponse({"success": False, "error": "No log files provided"})
            
            analysis = {
                "errors": [],
                "warnings": [],
                "communication_issues": [],
                "timing_issues": [],
                "memory_issues": []
            }
            
            for log_file in log_files:
                if os.path.exists(log_file):
                    with open(log_file, 'r', encoding='utf-8', errors='ignore') as f:
                        lines = f.readlines()
                    
                    for i, line in enumerate(lines):
                        line_lower = line.lower()
                        if any(word in line_lower for word in ['error', 'fail', 'exception', 'timeout']):
                            analysis['errors'].append({
                                "file": log_file,
                                "line": i + 1,
                                "content": line.strip()
                            })
                        elif any(word in line_lower for word in ['warning', 'caution']):
                            analysis['warnings'].append({
                                "file": log_file,
                                "line": i + 1,
                                "content": line.strip()
                            })
            
            return JSONResponse({"success": True, "analysis": analysis})
        except Exception as e:
            log(f"Error analyzing logs: {e}")
            return JSONResponse({"success": False, "error": str(e)}, status_code=500)

